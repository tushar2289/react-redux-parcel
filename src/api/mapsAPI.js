export async function getNearbyPlaces(type, center, map) {
  const location = new window.google.maps.LatLng(center.lat, center.lng);
  const request = {
    location,
    type,
    rankBy: window.google.maps.places.RankBy.DISTANCE,
  };

  const service = new window.google.maps.places.PlacesService(map);
  const places = await convertCallbackToAsyncAwait(service, request);
  const serializedPlaces = places.map(serializePlace);
  return serializedPlaces;
}

const convertCallbackToAsyncAwait = (service, request) => {
  return new Promise((resolve) => {
    service.nearbySearch(request, (r) => {
      resolve(r);
    });
  });
};

const serializePlace = (place) => {
  return {
    lat: place.geometry.location.lat(),
    lng: place.geometry.location.lng(),
    name: place.name,
    icon: place.icon,
    id: place.id,
    photo: place.photos?.length > 0 ? place.photos[0].getUrl() : null,
    rating: place.rating,
    types: place.types,
    user_ratings_total: place.user_ratings_total,
  };
};
