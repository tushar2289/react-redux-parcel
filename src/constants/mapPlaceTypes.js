export const RESTAURANT = 'restaurant';
export const BANK = 'bank';
export const HOSPITAL = 'hospital';
export const ATM = 'atm';
export const AIRPORT = 'airport';
export const TRAIN_STATION = 'train_station';
export const SUPERMARKET = 'supermarket';
export const GAS_STATION = 'gas_station';
export const SHOPPING_MALL = 'shopping_mall';
