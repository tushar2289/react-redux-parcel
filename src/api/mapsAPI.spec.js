import { RESTAURANT } from '../constants/mapPlaceTypes';
import { getNearbyPlaces } from './mapsAPI';

describe('Maps API', () => {
  it('should return the places', async () => {
    const places = await getNearbyPlaces(
      RESTAURANT,
      { lat: 100, lng: 100 },
      {}
    );
    expect(places).toMatchSnapshot();
  });
});
