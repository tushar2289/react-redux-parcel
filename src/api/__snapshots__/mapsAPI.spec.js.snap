// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maps API should return the places 1`] = `
Array [
  Object {
    "icon": "icon",
    "id": "id",
    "lat": undefined,
    "lng": undefined,
    "name": "Mumbai",
    "photo": null,
    "rating": 5,
    "types": Array [
      "restaurant",
    ],
    "user_ratings_total": 666,
  },
]
`;
